### DISCUSSION TITLE
Can julia be run ?

### DISCUSSION THREAD

**Discussion by 573** [nix-community/nix-on-droid#262] [Q&A]
`nix shell github:NixOS/nixpkgs/master#julia_19 -c julia -E 1+1` just gives me:

> ReadOnlyMemoryError()

It [seems to use](https://github.com/NixOS/nixpkgs/pull/210685#issuecomment-1548207615) `buildFHSUserEnv` so this might be an obstacle or is this irrelevant ?

Do you have further hints on how to tackle or even just explain this ?

**Comment by 573** [comment]
Using `nix repl` I can at least initiate a `julia` session using `julia_19-bin` but there eventually I will hit `ReadOnlyMemoryError()` again:

> $ nix repl
:lf github:NixOS/nixpkgs
:u outputs.legacyPackages.aarch64-linux.julia_19-bin
nix-shell$ julia   # now interact with that repl
julia> using Pkg
julia> Pkg.add("IJulia")   # there we go again...

**Reply by MikiVanousek** [reply]
It runs, I can even open Pluto notebooks using the device browser.

**Comment by 573** [comment]
`nix shell github:NixOS/nixpkgs/de3ac24#julia_19-bin` with the latest nix-on-droid revision gets me to run a `julia` repl but trying subsequent statements the repl is not responding and can only be left via CTRL-C instead of CTRL+D.

**Reply by 573** [reply]
In fakedroid the problem is absent.

**Reply by 573** [reply]
Adding `--impure` as in `nix shell github:NixOS/nixpkgs/de3ac24#julia_19-bin --impure` does mitigate some of the issues, but only helps insofar some julia primitives work (`log(2)` and the likes) some not (`split("Hello, World", ",")` also CTRL-D still doesn't work).

Those that aren't working cause `ReadOnlyMemoryError()` on my hardware.

Using `julia_19` instead seems no option as the build is broken.

I would be curious if some of you have had more success in how they did it.

**Reply by 573** [reply]
Trying `nix build --no-link --accept-flake-config --impure --print-build-logs github:NixOS/nixpkgs#pkgsCross.aarch64-multiplatform.julia_19` now.

**Reply by 573** [reply]
See https://github.com/NixOS/nixpkgs/issues/249905

**Reply by 573** [reply]
Tracking further analysis [here](https://gist.github.com/573/ceff8d298cc5e592b16dda9115d1d088) for now.
Tl;dr Using some not all of julia possible via or variants of (`--help`):
```
nix shell --impure --expr 'with import (builtins.getFlake "/data/data/com.termux.nix/files/home/.nix-config").inputs.unstable {}; julia' -c julia --check-bounds=yes --compile=no
```
